src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -642,17 +642,19 @@ public static double getPlayerVisibilityDistance(EntityPlayer player, double xzD
         return value >= maxXZDistance ? maxXZDistance : value;
     }
 
-    public static boolean isLivingOnLadder(@Nonnull IBlockState state, @Nonnull World world, @Nonnull BlockPos pos, @Nonnull EntityLivingBase entity)
+    public static boolean isLivingOnLadder(@Nonnull IBlockState state, @Nonnull World world, @Nonnull BlockPos pos, @Nonnull EntityLivingBase entity, EnumFacing side)
     {
         boolean isSpectator = (entity instanceof EntityPlayer && ((EntityPlayer)entity).isSpectator());
         if (isSpectator) return false;
         if (!ForgeModContainer.fullBoundingBoxLadders)
         {
-            return state.getBlock().isLadder(state, world, pos, entity);
+        	Block b1 = state.getBlock();
+        	Block b2 = world.getBlockState(pos.offset(side)).getBlock();
+        	return b1.isLadder(state, world, pos, entity) || b1.isLadder(state, world, pos, entity, side) || b2.isLadder(state, world, pos, entity) || b2.isLadder(state, world, pos, entity, side);
```

Comment:
The main difference lies here as far as I can tell, this would save you from calling an deprecated function and remove 2 checks